### ISSUE TITLE
[Bug] - database "kasmapp" does not exist same issue like in ->#331 |

### ISSUE THREAD

**Issue by hox-hub** [kasmtech/workspaces-issues#625]
### Existing Resources

- [X] Please search the existing issues for related problems
- [X] Consult the product documentation : [Docs](https://kasmweb.com/docs/latest/index.html)
- [X] Consult the FAQ : [FAQ](https://kasmweb.com/docs/latest/FAQ.html)
- [X] Consult the Troubleshooting Guide : [Guide](https://kasmweb.com/docs/latest/guide/troubleshooting.html)
- [X] Reviewed existing training videos: [Youtube](https://www.youtube.com/@KasmWorkspaces )

**Describe the bug**
I have installed KASM according to the instructions. (Meanwhile the 3rd time completely new on different OS and the error remains the same)

First of all, everything works completely as it should. For the first time! -After a few days, then every few hours, when I navigate to the site again, I only get an empty KASM background image and can do nothing. A look at the DEV tools shows that I get a 503 from almost every source. If I try to log in from another browser (or inprivate), I get an “incorrect password” (even with the correct PW). Again, this is because no service other than NGINX is actually running.

A look at the logs of the individual services shows a healthcheck error for “kasmweb/agent:1.15.0”.
 The same applies to “kasmweb/manager:1.15.0”. The other containers look normal.

A look at the logs of the Postgres DB within the container shows that the following message appears every few seconds: “database ‘kasmapp’ does not exist”

When I restart the server everything works again but only for a few hours (maximum always about 24).

Unfortunately, a solution was never mentioned in Issue #331.

**To Reproduce**
Wait ammount of time specified

**Expected behavior**
The service remains operational.

**Screenshots**
I can provide screenshots at a later date. Unfortunately, I restarted the service earlier in my own troubleshooting attempt and it now takes a few hours for it to fail again.

**Workspaces Version**
Version 1.15

**Workspaces Installation Method**
Single Server on VM (Proxmox Hypervisor)

**Client Browser (please complete the following information):**
All Browsers

**Workspace Server Information (please provide the output of the following commands):**
 - `Linux portal 6.1.0-25-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.106-3 (2024-08-26) x86_64 GNU/Linux`
 - `PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"`
 - `Client: Docker Engine - Community
 Version:    27.3.0
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.17.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.5.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-compose

Server:
 Containers: 14
  Running: 8
  Paused: 0
  Stopped: 6
 Images: 21
 Server Version: 27.3.0
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc version: v1.1.14-0-g2c9f560
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.1.0-25-amd64
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 11.68GiB
 Name: portal
 ID: ca60da3e-c9b0-432a-8646-c5f3e09ca509
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled`
 - `e78c05c87e21   kasmweb/nginx:1.25.3       "/docker-entrypoint.…"   3 hours ago   Up 52 minutes             80/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp   kasm_proxy
02342bb68f38   kasmweb/agent:1.15.0       "/bin/sh -c '/usr/bi…"   3 hours ago   Up 52 minutes (healthy)   4444/tcp                                        kasm_agent
e20c29bc1f82   kasmweb/share:1.15.0       "/bin/sh -c '/usr/bi…"   3 hours ago   Up 52 minutes (healthy)   8182/tcp                                        kasm_share
c64b1b783ed1   kasmweb/manager:1.15.0     "/bin/sh -c '/usr/bi…"   3 hours ago   Up 22 minutes (healthy)   8181/tcp                                        kasm_manager
52a591d47749   redis:5-alpine             "docker-entrypoint.s…"   3 hours ago   Up 52 minutes             6379/tcp                                        kasm_redis
9ac58bafa359   kasmweb/kasm-guac:1.15.0   "/dockerentrypoint.sh"   3 hours ago   Up 52 minutes (healthy)                                                   kasm_guac
6793afa64886   kasmweb/api:1.15.0         "/bin/sh -c '/usr/bi…"   2 days ago    Up 52 minutes (healthy)   8080/tcp                                        kasm_api
f70e9b4f83e2   postgres:12-alpine         "docker-entrypoint.s…"   2 days ago    Up 52 minutes (healthy)   5432/tcp                                        kasm_db`

**Comment by hox-hub** [comment]
I have reinstalled the software exactly according to the instructions including 10 GB swap with version 16 on a new VM and the behavior is the same again. It is so infinitely frustrating.

**Comment by j-travis** [comment]
This sounds like an issue with your environment as we've not seen any reports like this over the years. 
Even a clean install of 1.16 on a brand new VM  ... The install initially works but some time the issues start to occur?

Please provide the logs of the kasm_db container shortly before and including when the problem starts.

Provide the output of the following commands on the clean VM install

```
df
uname -a
sudo docker info
sudo docker volume ls
sudo docker ps -a
```

I'd also try to look for any system level logs (dmesg, kern.log , syslog) to see if any issues occur around the time this tarts to happen.

Do you have any other software that interacts with the VM like a backup system / live migration etc that may be causing corruption.
Do you have any failed or failing disks in your hypervisor

Where did you acquire your VM Iso? perhaps try ubuntu 22.04 LTS  and ensure Kasm is the only thing running on that VM

What kind of workspaces are you running initially within the system?

**Comment by hox-hub** [comment]
Thank you very much for your answer.

The advice to check my system and especially the hard disks was really good. I discovered a defective disk in a ZFS storage, which caused the IO delay of my host to go very high with every backup job. After I fixed this problem, the service has now been running for 3 days without any errors. I would observe the whole thing for another week and close the case if it no longer occurs. From the initial behavior, however, I would say that was the problem.

It is interesting that no other service had a problem with it. Am I right in concluding that fast hard disks and short IO times are crucial for the stability of KASM?

Thank you very much and best regards!
